.gitignore

```diff
@@ -1,25 +1,45 @@
-*.pyc
```

Comment:
I decided to tidy up the `.gitignore` file.

These files tend to clog up with a bunch of seemingly random entries, which can make it hard to reason about later on.

I grouped these into some headings, for clarity, and added some notes.

Reply:
Awesome 👍